src/resspect/classifier_registry.py

```diff
@@ -0,0 +1,44 @@
+from resspect.plugin_utils import get_or_load_class
+from resspect.classifiers import ResspectClassifer
+
+__all__ = ["CLASSIFIER_REGISTRY", "register_builtin_classifiers", "fetch_classifier_class"]
+
+CLASSIFIER_REGISTRY = {}
+
+def register_builtin_classifiers():
```

Comment:
For my own knowledge, I'm curious what the pros and cons of using a register function are compared to tracking the information in a class variable and registering in `__init_subclass__`.

```
    def __init_subclass__(cls, **kwargs):
        # Register all subclasses in a dictionary mapping class name to the
        # class object, so we can programmatically create objects from the names.
        super().__init_subclass__(**kwargs)
        cls.generators[cls.__name__] = cls
```

Or is there no real preference?

Reply:
I simply haven't used `__init_subclass__`, I believe that they will yield the same results. Though one pro to using this approach is that it wouldn't be necessary to execute any code in the top level `__init__` method to create the register.

Thanks for the suggestion, let me experiment with this a bit.

Reply:
Ok, yes, that approach is quiet nice. Thank you again for the suggestion. I'll update the code to use `__init_subclass__` instead of the register_builtin_classifiers function.